server/src/main/java/org/apache/druid/discovery/BaseNodeRoleWatcher.java

```diff
@@ -352,6 +345,24 @@ public void resetNodes(Map<String, DiscoveryDruidNode> fullNodes)
     }
   }
 
+  static void scheduleTimeout(
```

Comment:
This works, but it'd be nicer as a non-static method, so the caller doesn't need to pass in the executor. The watcher has the `listenerExecutor` already, so as a non-static this would be:

```java
void scheduleTimeout(long timeoutSeconds)
{
    listenerExecutor.schedule(
        this::cacheInitializedTimedOut,
        timeout,
        TimeUnit.SECONDS
    );
}
```

Reply:
Done